catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardBackupFileStorage.java

```diff
@@ -0,0 +1,210 @@
+        }
+
+        Path metacardPath = getMetacardDirectory(id);
+        if (metacardPath == null) {
+            String message = String.format("Unable to create metacard path directory for %s", id);
+            LOGGER.debug(message);
+            throw new MetacardBackupException(message);
+        }
+
+        try {
+            Path parent = metacardPath.getParent();
+            if (parent != null) {
+                Files.createDirectories(parent);
+            }
+            Files.createFile(metacardPath);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to create backup file {}.  File may already exist.",
+                    metacardPath,
+                    e);
+        }
+
+        try (OutputStream outputStream = new FileOutputStream(metacardPath.toFile())) {
+            IOUtils.write(data, outputStream);
+        }
+    }
+
+    public void setOutputDirectory(String outputDirectory) {
+        this.outputDirectory = outputDirectory;
+    }
+
+    public String getOutputDirectory() {
+        return outputDirectory;
+    }
+
+    public void refresh(Map<String, Object> properties) {
+        Object outputDirectory = properties.get(OUTPUT_DIRECTORY_PROPERTY);
+        if (outputDirectory instanceof String && StringUtils.isNotBlank((String) outputDirectory)) {
+            this.outputDirectory = (String) outputDirectory;
+            LOGGER.debug("Updating {} with {}", OUTPUT_DIRECTORY_PROPERTY, outputDirectory);
+        }
+
+        Object id = properties.get(ID);
+        if (id instanceof String) {
+            setId((String) id);
+        }
+    }
+
+    private void deleteBackupIfPresent(String filename) throws MetacardBackupException {
+        Path metacardPath = getMetacardDirectory(filename);
+        if (metacardPath == null) {
+            LOGGER.trace("Unable to delete backup for: {}", filename);
+            throw new MetacardBackupException("Unable to delete backup");
+        }
+
+        try {
+            Files.deleteIfExists(metacardPath);
+            while (metacardPath.getParent() != null && !metacardPath.getParent()
+                    .toString()
+                    .equals(outputDirectory)) {
+                metacardPath = metacardPath.getParent();
+                if (isDirectoryEmpty(metacardPath)) {
+                    FileUtils.deleteDirectory(metacardPath.toFile());
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.trace("Unable to delete backup file {}", metacardPath, e);
+            throw new MetacardBackupException("Unable to delete backup file", e);
+        }
+    }
+
+    public Path getMetacardDirectory(String id) {
+        if (StringUtils.isEmpty(id)) {
+            return null;
+        }
+
+        if (id.length() < 6) {
+            id = StringUtils.rightPad(id, 6, "0");
+        }
+
+        try {
+            return Paths.get(outputDirectory, id.substring(0, 3), id.substring(3, 6), id);
```

Comment:
These are used to split up the id into 2 directories. I think what is going on will be less clear getting into substrings and stuff. 6 is magic on because you need at least 6 characters to split into two 3 character substrings.